ops/roles/infra/node/tasks/main.yml

```diff
@@ -1,21 +1,26 @@
 ---
-- name: Install required packages
-  apt:
-    name: curl
-    update_cache: yes
-    state: latest
+- name: Create keyrings directory
+  ansible.builtin.file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: "755"
 
-- name: Add the NodeSource repository to the system
-  shell: curl -sL https://deb.nodesource.com/setup_16.x | sudo bash -
+- name: Download and import the Nodesource GPG key
+  ansible.builtin.shell: set -o pipefail && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+  args:
+    executable: /bin/bash
 
-- name: Install NodeJS and Npm
-  apt:
-    name: nodejs
-    update_cache: yes
-    state: latest
+- name: Create deb repository
+  ansible.builtin.shell:
+    set -o pipefail && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list # Remember to update the major version of NPM when updating the major version of Node
+  args:
+    executable: /bin/bash
 
-- name: Update Npm to latest version
-  command: npm install -g npm
+- name: Install NodeJS and NPM
+  ansible.builtin.apt:
+    name: nodejs
+    update_cache: true
+    state: latest # The major version of NodeJS is provided by the NodeSource repository defined in the task above
 
-- name: Install pm2
-  command: npm install -g pm2 --production=true
```

Comment:
Why was this removed? Is this now working by coincidence because `pm2` is already installed as a leftover?

Reply:
That's a good point! Here is the fix: https://github.com/OpenTermsArchive/contribution-tool/pull/157